dotnet/Vaas/src/Vaas/Vaas.cs

```diff
@@ -0,0 +1,181 @@
+            {
+                if (msg.MessageType != WebSocketMessageType.Text) return;
+                var message = JsonSerializer.Deserialize<Message>(msg.Text);
+                switch (message?.Kind)
+                {
+                    case "AuthResponse":
+                        var authenticationResponse = JsonSerializer.Deserialize<AuthenticationResponse>(msg.Text);
+                        if (authenticationResponse is {Success: true})
+                        {
+                            Authenticated = true;
+                            SessionId = authenticationResponse.SessionId;
+                        }
+                        break;
+                        
+                    case "VerdictResponse":
+                        var options = new JsonSerializerOptions {Converters = {new JsonStringEnumConverter()}};
+                        var verdictResponse = JsonSerializer.Deserialize<VerdictResponse>(msg.Text,options);
+                        VerdictResponsesDict.Add(verdictResponse?.Guid ?? throw new InvalidOperationException(), verdictResponse);
+                        break;
+                }
+            });
+            Client.Start().GetAwaiter().GetResult();
+            if (!Client.IsStarted)
+            {
+                throw new WebsocketException("Could not start client");
+            }
+            
+            Authenticate();
+            Task.Run(SendPing);
+        }
+
+        private void Authenticate()
+        {
+            var authenticationRequest = new AuthenticationRequest(Token);
+            var jsonString = JsonSerializer.Serialize(authenticationRequest);
+            Client?.Send(jsonString);
+            for (var i = 0; i < 10; i++)
+            {
+                if (Authenticated)
+                {
+                    break;
+                }
+
+                Thread.Sleep(100);
+            }
+
+            if (Authenticated != true)
+            {
+                throw new UnauthorizedAccessException();
+            }
+        }
+
+        public async Task<Verdict> ForSha256Async(string sha256)
+        {
+            var value = await ForRequestAsync(new AnalysisRequest(sha256,SessionId ?? throw new InvalidOperationException()));
+            return value.Verdict;
+        }
+        
+        public async Task<Verdict> ForFileAsync(string path)
+        {
+            var sha256 = Sha256CheckSum(path);
+            var verdictResponse = await ForRequestAsync(new AnalysisRequest(sha256, SessionId ?? throw new InvalidOperationException()));
+            if (verdictResponse.Verdict != Verdict.Unknown) return verdictResponse.Verdict;
+            var url = verdictResponse.Url;
+            if (url is null) throw new ArgumentNullException(nameof(url));
+            
+            var token = verdictResponse.UploadToken;
+            var data = await File.ReadAllBytesAsync(path);
+            using (var client = new WebClient())

```

Comment:
WebClient is obsolete. The modern version is HttpClient. Make it a field, otherwise it opens a new socket for every request.